src/uct/ib/rc/base/rc_ep.h

```diff
@@ -41,17 +73,44 @@ enum {
         return UCS_ERR_NO_RESOURCE; \
     }
 
+/*
+ * check for FC credits and add FC protocol bits (if any)
+ */
+#define UCT_RC_CHECK_FC_WND(_iface, _ep, _am_id) \
+    do { \
+        if ((_ep)->fc_wnd <= (_iface)->config.fc_soft_thresh) { \
+            if ((_ep)->fc_wnd <= 0) { \
+                return UCS_ERR_NO_RESOURCE; \
+            } \
+            (_am_id) |= uct_rc_ep_fc_req_moderation(_iface, _ep); \
+        } \
+        (_am_id) |= uct_rc_ep_get_fc_hdr((_ep)->flags); /* take grant bit */ \
```

Comment:
We can, but we have 7 free bits in flags field now. We will have to mask fc portion anyway when start using these free bits for some purpose. And it takes only one "and" bit operation